crates/wasmtime/src/runtime/component/func/host.rs

```diff
@@ -238,6 +266,7 @@ unsafe fn call_host<T, Params, Return, F>(
     ty: TypeFuncIndex,
     options: OptionsIndex,
     storage: &mut [MaybeUninit<ValRaw>],
+    async_function: bool,
```

Comment:
Could this be plumbed through with `S` as a type parameter rather than an extra argument to follow the same idioms of the other constructors in this file?